apis/bigqueryconnection/v1alpha1/connection_types.go

```diff
@@ -38,8 +38,11 @@ type Parent struct {
 type BigQueryConnectionConnectionSpec struct {
 	Parent `json:",inline"`
 
-	// The BigQuery ConnectionID. This is a server-generated ID in the UUID format.
-	// If not provided, ConfigConnector will create a new Connection and store the UUID in `status.serviceGeneratedID` field.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
```

Comment:
There's a potential problem for optional and immutable check on the CRD level, where  "" --> "<connection ID>" is considered invalid. Can we remove the immutable check here? I think it's safe to remove this check because  the NewRef function validates the resourceID  as well.

Reply:
Removed.